pilot/pkg/bootstrap/galley.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2019 Istio Authors
```

Comment:
Why is this file moving into Pilot? Also, what happens if I want to use Pilot as is, without Galley? This will force me to use galley code always, when I dont want to.

Reply:
There will not be an option - the plan is to remove the duplicated code reading configs from k8s.

Not enabling Galley validation or k8s is still possible.

This is what we discussed in the env meeting ( John's proposal ) - to link all components in pilot.

Reply:
I would rather add a wrapper function for all depended components in istiod instead of bringing the risk of breaking a stable controlplane

Reply:
I don't see how it would break - we are shipping with MCP as default, the code is just running both in same process to avoid startup deps issues.

Reply:
Removed for now, to unblock the testing and rest of the PR